highfidelity/hifi
# Android - App freezes on opening randomly - Fix

Previous version is having the following bug:
- After closing the app, starting it again can lead to the loading screen that doesn't advance, is frozen/crashed. It required Clearing app data to run again.
- After fixing that problem, a similar bug happened but with a black screen instead of the loading screen.
- Both problems should be solved and a quick-to-appear splash screen is shown. Our traditional loading screen adds a loading indicator.

**Test plan**

1. Start the application.
2. High Fidelity splash screen should Immediately appear (no more black screen).
3. [Technical note - When Interface starts loading] A loading indicator should appear.
4. App main screen with list of domains should appear as always.
5. Close the app by pressing back (hardware button or android bottom bar button). **Next steps are critical as there we had our crash/freeze.**
6. Launch Interface again.
7. App should load as before. (In contrast with our two bugs: Stuck on Splash screen nor Black screen).
8. Exit the app by pressing home (hardware button or android bottom bar button).
9. Launch Interface again.
10. In normal memory conditions, the app should start where we left (in this case the Domains list).

Note: There is another issue: A crash that can happen when returning from a domain. That will be addressed separately.
